### PR TITLE
docs: add Merge Protections docs

### PR DESCRIPTION
**Pull request by jd**
Change-Id: I48852b0e93c3e89370bb12188f95cd192492d1f7

### CODE REVIEW

**Comment by jd** [comment]
This pull request is part of a stack:
1. docs: add Merge Protections docs ([#4659](https://github.com/Mergifyio/docs/pull/4659)) 👈

**Review by sileht — APPROVED** [review]

File: src/content/docs/merge-protections/index.mdx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import scheduleFreezeScreen from "../../images/merge-protections/schedule-freeze.png"
+import commentScreen from "../../images/merge-protections/schedule-freeze.png"
+
+[GitHub's branch
+protection](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches)
+and
+[rulesets](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/about-rulesets)
+are foundational, but their customizations are somewhat limited.
+
+For more advanced or specific scenarios, Mergify's Merge Protection can step
```
Comment (Greesb) [inline src/content/docs/merge-protections/index.mdx:21]:
```suggestion
For more advanced or specific scenarios, Mergify's Merge Protections can step
```

**Review by Syffe — COMMENTED** [review]
otherwise all good!

File: src/content/docs/merge-protections/index.mdx (hunk 1 above)
Comment (jd) [reply src/content/docs/merge-protections/index.mdx:21]: fixed

File: src/content/docs/merge-protections/index.mdx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+Once enabled on your repository, Mergify will post a check entitled `Mergify
+Merge Protections` on your pull requests. You need to ensure this checks is
+required in your branch protection settings or in your rulesets.
+
+<Image src={bpScreen} alt="GitHub Branch Protection" />
+
+Once `Mergify Merge Protections` is required in your repository rules, it will
+be marked as enforced in your GitHub checks and will prevent any pull request
+to be merged if does not matches your conditions.
```
Comment (DouglasBlackwood) [inline src/content/docs/merge-protections/index.mdx:55]:
```suggestion
to be merged if it does not match your conditions.
```

**Review by DouglasBlackwood — DISMISSED** [review]

**Review by Greesb — DISMISSED** [review]

**Review by sileht — APPROVED** [review]

**Review by Syffe — APPROVED** [review]

**Review by DouglasBlackwood — APPROVED** [review]

**Pull request merged**